falcon/request.py

```diff
@@ -1773,9 +1773,10 @@ class RequestOptions(object):
                 encoded according to the standard W3C algorithm (see
                 also http://goo.gl/6rlcux).
 
-        auto_parse_qs_csv: Set to ``False`` to treat commas in a query
-            string value as literal characters, rather than as a comma-
-            separated list (default ``True``). When this option is
+        auto_parse_qs_csv: Set to ``True`` to split on any
+            non-percent-encoded commas.
```

Comment:
Sorry for a bit of a late finding, but the sentence seems to be somehow broken here with a full stop, and continued in lower-case in a new line afterwards.

Reply:
I am probably too picky... but I would still like to polish this docstring even further :slightly_smiling_face: 

Right now it reads:
> Set to ``True`` to split on any non-percent-encoded commas. As a comma separated list, rather than a string value as literal characters (default ``False``).

The sentence got a bit fragmented, maybe we could join it into one sentence or rewrite in some other way?

Reply:
@MichaelYusko How about:

> Set to ``True`` to split a query string value on any non-percent-encoded commas.
> This option is disabled by default (``False``), and commas in the value will be treated as literal characters, rather than as a comma- separated list. The default behavior allows encoding lists as multiple occurrences of the same parameter, as well as cases when values may be encoded in alternative formats in which the comma character is significant.

@nZac suggestions are very welcome should you have any idea how to word this nicer :slightly_smiling_face:

Reply:
@vytas7 Yeah, I like it! and maybe we will wait an answer from @nZac , maybe he will have some suggestions.

Reply:
> Pass in `True` to split a query string value on any non-percent-encoded commas, this option is disabled by default (`False`). 

^ This looks good, keep it as the first line, below are two alternate phrasings. If we can give it a little white space, it might help readability, but I am not sure how to format that so renders correctly.

> When `False` query strings containing commas are untouched, this enables encoding lists as multiple occurrences of the same parameter and cases where the comma character is significant such as alternate encodings (i.e. `/?t=1,2,3&t=4` becomes `['1,2,3', '4']`).
> 
> When `True`, the query string value is split on the commas and returned as a list (i.e. `/?t=1,2,3&t=4` becomes `['1', '2', '3', '4']`). 


-- or --

> Consider this query string `/?t=1,2,3&t=4` and the impact on `req.params`. When `auto_parse_qs_csv` is `True` it is represented as `{ 't': ['1', '2', '3', '4'] }`, when `False` it will be `{ 't': ['1,2,3', '4'] }`. 
> 
> While auto-parsing can be a helpful shortcut, the default of `False` works better with alternative encodings and representing multiple parameters with the same key.

-- or --

You can use the previous one 😄.

Reply:
@vytas7 @nZac I prefer this one, because documentation with some examples is always helpful.

```
When False query strings containing commas are untouched, this enables encoding lists as multiple occurrences of the same parameter and cases where the comma character is significant such as alternate encodings (i.e. /?t=1,2,3&t=4 becomes ['1,2,3', '4']).
```